### DISCUSSION TITLE
AnalyticalTable: Filter menu closed on keypress ENTER

### DISCUSSION THREAD

**Discussion by reqyou** [UI5/webcomponents-react#4337] [Q&A] [answered]
Hey guys 👋,
I am currently facing the issue that the `AnalyticalTable` `Filter Menu` gets closed once I hit ENTER. The problem is the following. I've built a custom filter with a `DateRangePicker` in it, whose value I can change manually. I would expect that I am able to submit the DateRangePicker value with ENTER, but the Filter Menu is closed before submission. See also the demo I've attached. For me personally this behavior does not make a lot of sense.

Does anyone know how to prevent closing the Filter Menu on press ENTER? Looks like that has also nothing to do with the tanstack react-table implementation.

Thanks guys!!

Best,
reQ

![GIF 14 03 2023 19-50-17](https://user-images.githubusercontent.com/101811919/225108961-1db3f774-864b-4d9d-af42-6788782c9588.gif)

**Comment by Lukas742** [comment, marked as answer]
Hi @reqyou 

the `keydown` event of the DateRangePicker is [bubbling up](https://en.wikipedia.org/wiki/Event_bubbling) to the popover. So to prevent closing the popover you can call `event.stopPropagation` in the respective event handler.

https://codesandbox.io/p/sandbox/at-custom-column-filter-forked-1shce7?file=%2Fsrc%2FApp.js

**Reply by reqyou** [reply]
Works like charm!! Thank you!
